crates/starknet/src/starknet/dump.rs

```diff
@@ -0,0 +1,178 @@
+                    file.set_len(lenght - 1).map_err(Error::IoError)?; // remove last "]" with set_len
+                    file.write_all(format!(", {transaction_dump}]").as_bytes())
+                        .map_err(Error::IoError)?;
+                } else {
+                    // create file
+                    let transactions = vec![transaction];
+                    let transactions_dump = serde_json::to_string(&transactions).map_err(|_| {
+                        Error::SerializationError { obj_name: "Vec<Transaction>".to_string() }
+                    })?;
+                    fs::write(Path::new(&path), transactions_dump)?;
+                }
+
+                Ok(())
+            }
+            None => Err(Error::FormatError),
+        }
+    }
+
+    /// save starknet transactions to file
+    pub fn dump_transactions(&self) -> DevnetResult<()> {
+        match &self.config.dump_path {
+            Some(path) => {
+                let transactions = &self
+                    .transactions
+                    .iter()
+                    .map(|x| x.1.inner.clone())
+                    .collect::<Vec<Transaction>>();
+
+                // dump only if there are transactions to dump
+                if !transactions.is_empty() {
+                    let transactions_dump = serde_json::to_string(transactions).map_err(|_| {
+                        Error::SerializationError { obj_name: "Vec<Transaction>".to_string() }
+                    })?;
+                    fs::write(Path::new(&path), transactions_dump)?;
+                }
+
+                Ok(())
+            }
+            None => Err(Error::FormatError),
+        }
+    }
+
+    // load starknet transactions from file
+    pub fn load_transactions(&self) -> DevnetResult<Vec<Transaction>> {
+        match &self.config.dump_path {
+            Some(path) => {
+                let file_path = Path::new(path);
+
+                // load only if the file exists, if dump_path is set but the file doesn't exist it
+                // can mean that it's first run with dump_path parameter set to dump, in that case
+                // return empty vector
+                if file_path.exists() {
+                    let mut file = File::open(file_path).map_err(Error::IoError)?;
+                    let mut data = String::new();
+                    file.read_to_string(&mut data).map_err(Error::IoError)?;
+                    let transactions: Vec<Transaction> =
+                        serde_json::from_str(&data).map_err(|_| Error::DeserializationError {
+                            obj_name: "Vec<Transaction>".to_string(),
+                        })?;
+
+                    // to avoid doublets in transaction mode during load, we need to remove the file
+                    // because they will be re-executed and saved again
+                    if self.config.dump_on == Some(DumpMode::OnTransaction) {
```

Comment:
although it's fixing an issue I'm not sure if I'm satisfied with that, in the other hand detecting that in handle_successful_transaction might be tricky, I'm still checking this.

Reply:
I don't understand, which "doublets" does the code comment refer to?

Reply:
I understand now, upon starting Devnet with a path to load from, re-execution would lead to re-addition of txs into the dump file. It really seems that removing the file the simplest way. Keeping this conversation open to hear other opinions.